### PR TITLE
Add CI using Github actions

### PR DESCRIPTION
**Pull request by gwynne**
Based heavily on Vapor's configuration

### CODE REVIEW

**Comment by 0xTim** [comment]
@gwynne for `master` we should use the 5.1 images since they include OpenSSL

**Comment by gwynne** [comment]
CircleCI config will need to be removed from this repo's config if/when this is merged.

@0xTim Whoops, didn't realize this PR was targeted to the wrong base branch!

**Comment by gwynne** [comment]
Ugh, wish GH's UI could understand doing rebase-onto...

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+on:
+- pull_request
+jobs:
+  imperial_macos:
+    runs-on: macos-latest
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_11.4_beta.app/Contents/Developer
+    steps:
+    - uses: actions/checkout@v2
+    - run: brew install pkg-config libressl
```
Comment (0xTim) [inline .github/workflows/test.yml]: libressl is not required for Vapor 4

**Comment by 0xTim** [comment]
@gwynne will a `git commit --allow-empty` trigger CI?

File: .github/workflows/test.yml (hunk 1 above)
Comment (gwynne) [reply .github/workflows/test.yml]: See my comments over on the Vapor PR re: this

**Comment by gwynne** [comment]
> @gwynne will a `git commit --allow-empty` trigger CI?

@0xTim I think so.

**Comment by 0xTim** [comment]
Merging as it's just CircleCI that's failing now

**Pull request merged**